src/Storageless/Http/ClientFingerprint/UserAgent.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PSR7Sessions\Storageless\Http\ClientFingerprint;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+use function array_key_exists;
+use function is_string;
+
+/** @immutable */
+final class UserAgent implements Source
+{
+    public const REQUEST_ATTRIBUTE_NAME = 'HTTP_USER_AGENT';
+
+    public function extractFrom(ServerRequestInterface $request): string
+    {
+        $serverParams = $request->getServerParams();
```

Comment:
I don't think we should rely on `$_SERVER` naming and parameters. The header is called `User-Agent`: we should use that.